persistence/README.md

```diff
@@ -78,232 +114,306 @@ This document presents a specification for the Persistence Module of the Pocket
+        P4B: Pocket Core Process (ID B)
+        P4C: Pocket Core Process (ID C)
+        DB4: Database Engine Process
+
+        state join_state4 <<join>>
+        P4A --> join_state4
+        P4B --> join_state4
+        P4C --> join_state4
+        join_state4 --> DB4: IBC, TCP/IP, HTTP (local communications)
+    }
+```
 
-The selected **Database Engine** which will be referred to as “**the database engine**” will be [PostgreSQL](https://www.postgresql.org/). The accompanying research document to this Pre-Planning Specification contains the research by which this decision was made. Postgresql contains multiple desirable properties that satisfy requirements such as **Schema Definition Mechanism, Deterministic Write Mechanism **and** Idempotent Dataset Updates**.
+### 1.2. Database Engine
 
-### 1.2.1. Configuration
+The selected **Database Engine**, which will be referred to as “**the database engine**” will be [PostgreSQL](https://www.postgresql.org/). The accompanying research document to this Pre-Planning Specification contains the research through which this decision was made. In addition, PostgreSQL has multiple desirable properties that satisfy requirements such as **Schema Definition Mechanism**, **Deterministic Write Mechanism **and** Idempotent Dataset Updates**.
 
-When implementing a client-server architecture, several approaches can be used to establish communication between the middleware and the database engine. Below we describe a data-structure that contains the necessary attributes:
+#### 1.2.1. Configuration
 
+When implementing a client-server architecture, several approaches can establish communication between the middleware and the database engine. Below we describe a data structure that contains the necessary attributes:
 
 ```
 type DBConfiguration interface {
-  Host() URL           # host url where to contact the database engine e.g. 1.2.3.4, localhost
-  Username() String    # the username of the system user the middleware will use to access
-  Password() String    # the password of the system user the middleware will use to access
+	Host() URL           # host url where to contact the database engine e.g. 1.2.3.4, localhost
+	Username() String    # the username used by the middleware to access the database engine
+	Password() String    # the password used by the middleware to access the database engine
 }
 ```
 
-On the database engine the following configurations need to be made:
+On the database engine, the following configurations need to be specified:
+
+1. User role: 1 of **Servicer**, **Validator**, **Fisherman**, **Archival** or **Full Node**.
+2. Schemas: 1 or more of **Consensus**, **State**, **Mempool** and **Local**.
 
-1. User roles: **Servicer**, **Validator**, and **Full Nodes**.
-2. Schemas: **Consensus**, **State**, **Mempool** and **Local**.
+<!-- TODO(discuss in review): Are the changes here accurate? Should we also add block store and other types of schemas?-->
```

Comment:
The block store is part of the `Consensus` dataset.

Reply:
Makes sense. Resolving.